controllers/authorino_task.go

```diff
@@ -0,0 +1,121 @@
+package controllers
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	v1beta2 "github.com/kuadrant/authorino-operator/api/v1beta1"
+	"github.com/kuadrant/policy-machinery/controller"
+	"github.com/kuadrant/policy-machinery/machinery"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/utils/ptr"
+
+	"github.com/kuadrant/kuadrant-operator/api/v1beta1"
+)
+
+type AuthorinoCrReconciler struct {
+	Client *dynamic.DynamicClient
+}
+
+func NewAuthorinoCrReconciler(client *dynamic.DynamicClient) *AuthorinoCrReconciler {
+	return &AuthorinoCrReconciler{Client: client}
+}
+
+func (r *AuthorinoCrReconciler) Subscription() *controller.Subscription {
+	return &controller.Subscription{
+		ReconcileFunc: r.Reconcile,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: ptr.To(v1beta1.KuadrantKind), EventType: ptr.To(controller.CreateEvent)},
+			{Kind: ptr.To(v1beta1.AuthorinoKind), EventType: ptr.To(controller.DeleteEvent)},
+		},
+	}
+}
+
+func (r *AuthorinoCrReconciler) Reconcile(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, _ *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("AuthorinoCrReconciler")
+	logger.Info("reconciling authorino resource", "status", "started")
+	defer logger.Info("reconciling authorino resource", "status", "completed")
+
+	kobjs := lo.FilterMap(topology.Objects().Roots(), func(item machinery.Object, _ int) (*v1beta1.Kuadrant, bool) {
+		if item.GroupVersionKind().Kind == v1beta1.KuadrantKind.Kind {
+			return item.(*v1beta1.Kuadrant), true
+		}
+		return nil, false
+	})
+
+	kobj, err := GetOldestKuadrant(kobjs)
+	if err != nil {
+		if strings.Contains(err.Error(), "empty list passed") {
+			logger.Info("kuadrant resource not found, ignoring", "status", "skipping")
+			return err
+		}
+		logger.Error(err, "cannot find Kuadrant resource", "status", "error")
+		return err
+	}
+
+	aobjs := lo.FilterMap(topology.Objects().Objects().Children(kobj), func(item machinery.Object, _ int) (machinery.Object, bool) {
```

Comment:
nit: redundant consecutive `.Objects()` call? 🤔  
```suggestion
	aobjs := lo.FilterMap(topology.Objects().Children(kobj), func(item machinery.Object, _ int) (machinery.Object, bool) {
```